src/components/inputs/ui-select-extended/ui-select-extended.tsx

```diff
@@ -0,0 +1,154 @@
+import { OHRIFormFieldProps } from '../../../api/types';
+import { useField } from 'formik';
+import styles from './ui-select-extended.scss';
+import { OHRIFormContext } from '../../../ohri-form-context';
+import { getConceptNameAndUUID } from '../../../utils/ohri-form-helper';
+import { OHRIFieldValueView } from '../../value/view/ohri-field-value-view.component';
+import { isTrue } from '../../../utils/boolean-utils';
+import { getDataSource } from '../../../registry/registry';
+import { fieldRequiredErrCode, isEmpty } from '../../../validators/ohri-form-validator';
+import { PreviousValueReview } from '../../previous-value-review/previous-value-review.component';
+import debounce from 'lodash-es/debounce';
+import InlineLoader from '../../loaders/inline-loader.component';
+
+export const UISelectExtended: React.FC<OHRIFormFieldProps> = ({ question, handler, onChange }) => {
+  const [field, meta] = useField(question.id);
+  const { setFieldValue, encounterContext, fields } = React.useContext(OHRIFormContext);
+  const [conceptName, setConceptName] = useState('Loading...');
+  const [items, setItems] = useState([]);
+  const [warnings, setWarnings] = useState([]);
+  const [errors, setErrors] = useState([]);
+  const [isLoading, setIsLoading] = useState(false);
+  const [searchTerm, setSearchTerm] = useState('');
+  const isFieldRequiredError = useMemo(() => errors[0]?.errCode == fieldRequiredErrCode, [errors]);
+  const [previousValueForReview, setPreviousValueForReview] = useState(null);
+  const inputValue = useRef('');
+  const dataSource = useMemo(() => getDataSource(question.questionOptions['datasource']), []);
+  useEffect(() => {
+    if (question['submission']) {
+      question['submission'].errors && setErrors(question['submission'].errors);
+      question['submission'].warnings && setWarnings(question['submission'].warnings);
+    }
+  }, [question['submission']]);
+
+  const handleChange = value => {
+    setFieldValue(question.id, value);
+    onChange(question.id, value, setErrors, setWarnings);
+    question.value = handler?.handleFieldSubmission(question, value, encounterContext);
+  };
+
+  const debouncedSearch = debounce((searchterm, dataSource) => {
+    setIsLoading(true);
+    dataSource.fetchData(searchterm).then(dataItems => {
+      setItems(dataItems.map(dataSource.toUuidAndDisplay));
+      setIsLoading(false);
+    });
+  }, 300);
+
+  useEffect(() => {
+    // If not searchable, preload the items
+    if (dataSource && !isTrue(question.questionOptions['isSearchable'])) {
+      setIsLoading(true);
+      dataSource.fetchData().then(dataItems => {
+        setItems(dataItems.map(dataSource.toUuidAndDisplay));
+        setIsLoading(false);
+      });
+    }
+  }, [dataSource]);
+
+  useEffect(() => {
+    // get the data source
+    if (dataSource && isTrue(question.questionOptions['isSearchable']) && !isEmpty(searchTerm)) {
+      debouncedSearch(searchTerm, dataSource);
+    } else {
```

Comment:
@kajambiya did you forget to handle this?

Reply:
> kajambiya

kachi???? 😊